src/dlr_tvm.cc

```diff
@@ -198,3 +207,90 @@ void TVMModel::UseCPUAffinity(bool use) {
     LOG(INFO) << "CPU Affinity is disabled";
   }
 }
+
+bool TVMModel::HasMetadata() const {
+  if (this->metadata != nullptr) {
+    return true;
+  }
+  return false;
+}
+
+void TVMModel::GetOutputNames(char**& names) const {
+  names = (char**) malloc(this->num_outputs_);
+  std::string name;
+
+  for(int i = 0; i < this->num_outputs_; i++) {
+    name = this->metadata["Model"]["Outputs"][i]["name"];
+    names[i] = (char *)malloc((name.size() + 1)*sizeof(char));
+    strcpy(names[i], name.c_str());
+  }
+}
+
+void TVMModel::GetInputNames(char**& names) const {
+  names = (char**) malloc(this->num_inputs_);
+  std::string name;
+
+  for(int i = 0; i < this->num_inputs_; i++) {
+    name = this->metadata["Model"]["Inputs"][i]["name"];
+    names[i] = (char *)malloc((name.size() + 1)*sizeof(char));
+    strcpy(names[i], name.c_str());
+  }
+}
+
+void TVMModel::GetOutputIndex(const char* name, int* index) const {
+  char** output_names;
+  this->GetOutputNames(output_names);
+  int idx = -1;
+
+  for (int i = 0; i < this->num_outputs_; i++) {
+    if (strcmp(name, output_names[i]) == 0) {
+      idx = i;
+      break;
+    }
+  } 
+  *index = idx;
+}
+
+void TVMModel::GetInputIndex(const char* name, int* index) const {
+  char** input_names;
+  this->GetInputNames(input_names);
+  int idx = -1;
+
+  for (int i = 0; i < this->num_outputs_; i++) {
+    if (strcmp(name, input_names[i]) == 0) {
+      idx = i;
+      break;
+    }
+  } 
+  *index = idx;
+}
+
+void TVMModel::GetOuputByName(const char* name, float* out) {
+  int* output_index;
+  this->GetOutputIndex(name, output_index);
+  if (*output_index >= 0) {
+    this->GetOutput(*output_index, out);
+  }
+}
+
+void TVMModel::GetInputDataTypes(char**& dtypes) const {
+  dtypes = (char**) malloc(this->num_inputs_);
+  std::string dtype;
+
+  for(int i = 0; i < this->num_inputs_; i++) {
+    dtype = this->metadata["Model"]["Inputs"][i]["dtype"];
+    dtypes[i] = (char *)malloc((dtype.size() + 1)*sizeof(char));
+    strcpy(dtypes[i], dtype.c_str());
+  }
+}
+
+void TVMModel::GetOuputDataTypes(char**& dtypes) const {
+  dtypes = (char**) malloc(this->num_outputs_);
+  std::string dtype;
+
+  for(int i = 0; i < this->num_outputs_; i++) {
+    dtype = this->metadata["Model"]["Outputs"][i]["dtype"];
+    dtypes[i] = (char *)malloc((dtype.size() + 1)*sizeof(char));
+    strcpy(dtypes[i], dtype.c_str());
+  }
+}
```

Comment:
Unix txt file should end with End of line.

Reply:
👍